maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,255 @@
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
+			}
+		}
+
+		return !empty( $tagsToRemoveArrayKeyIds );
+	}
+
+	public function removeImapTagFromArticle( $articleId, $stringWithTagToRemove ) {
+		$user = User::newFromName( 'WikiaBot' );
+		$article = Article::newFromID( $articleId );
+
+		if ( $user->getId() === 0 ) {
+			echo 'Failed using WikiaBot user' . PHP_EOL;
+		}
+
+		if ( $article->getID() ) {
+			$oldContent = $article->getContent();
+			$newContent = str_replace( $stringWithTagToRemove, "", $oldContent );
+
+			self::debug( sprintf( 'Trying to edit article #%d', $articleId ) );
+			if( !self::isTest() ) {
+				$result = $article->doEdit(
+					$newContent,
+					'Real world maps have been discontinued by Wikia.',
+					EDIT_FORCE_BOT,
+					false,
+					$user
+				);
+			} else {
+				$result = true;
+			}
+
+
+			if( $result ) {
+				echo sprintf( "Removed <imap/> tags from article #%d", $articleId ) . PHP_EOL;
+			} else {
+				echo sprintf( "Failed to update article #%d", $articleId ) . PHP_EOL;
+			}
+		} else {
+			echo sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL;
+		}
+	}
+
+	public function isValidTilesSetId( $tilesSetId ) {
+		if ( !self::isValidInteger( $tilesSetId ) ) {
+			return false;
+		}
+
+		$res = $this->maps->getTileSet( $tilesSetId );
+
+		if ( !isset( $res['success'] ) || $res['success'] !== true ) {
+			self::debug( 'API call failure when looking for a tiles set #' . $tilesSetId . '.' );
+			die;
+		}
+
+		self::debug( "Tiles' set #" . $tilesSetId . " found." );
+		return $res['content']->id == $tilesSetId;
+	}
+
+	public function execute() {
+		$this->app = F::app();
+		$this->mapsClientConfig = $this->app->wg->IntMapConfig;
+		$this->maps = new WikiaMaps( $this->mapsClientConfig );
+
+		self::$test = $this->hasOption( 'test' );
+		self::$verbose = $this->hasOption( 'verbose' );
+
+		$cityId = $this->app->wg->CityId;
+		$tilesSetId = $this->getOption( 'tiles-set-id' );
+
+		if ( self::isTest() ) {
+			self::debug( 'Mode: test run' );
+		} else {
+			self::debug( 'Mode: normal run' );
+		}
+
+		if ( !self::isValidCityId( $cityId ) ) {
+			self::debug( 'Invalid city-id. Try again.' );
+			die;
+		}
+
+		if ( !$this->isValidTilesSetId( $tilesSetId ) ) {
+			self::debug( 'Invalid tiles-set-id. Try again.' );
+			die;
+		}
+
+		$mapsUsingTheTileset = self::getMapsIdsUsingTileset( $cityId, $tilesSetId );
+		$mapsUsingTheTilesetCount = count($mapsUsingTheTileset);
+		self::debug( sprintf( "Found %d maps using the tiles's set #%d", $mapsUsingTheTilesetCount, $tilesSetId ) );
+
+		if ( $mapsUsingTheTilesetCount === 0 ) {
+			echo 'No maps found.' . PHP_EOL;
+			die;
+		}
+
+		$articlesUsingImap = self::getArticlesIdsUsingImap( $cityId );
+		$articlesUsingImapCount = count( $articlesUsingImap );
+		self::debug( sprintf( "Found %d articles using <imap/>", $articlesUsingImapCount ) );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)